netplan/cli/commands/status.py

```diff
@@ -0,0 +1,656 @@
+        '''
+        Extract the MAC address if it's set inside the JSON data and seems to
+        have the correct format. Return 'None' otherwise.
+        '''
+        if 'address' in ip and len(ip['address']) == 17:  # 6 byte MAC (+5 colons)
+            return ip['address'].lower()
+        return None
+
+    def __init__(self, ip: JSON, nd_data: JSON = [], nm_data: JSON = [],
+                 resolved_data: tuple = (None, None), route_data: tuple = (None, None)):
+        self.idx: int = ip['ifindex']
+        self.name: str = ip['ifname']
+        self.adminstate: str = 'UP' if 'UP' in ip['flags'] else 'DOWN'
+        self.operstate: str = ip['operstate'].upper()
+        self.macaddress: str = self.__extract_mac(ip)
+
+        # Filter networkd/NetworkManager data
+        nm_data = nm_data or []  # avoid 'None' value on systems without NM
+        self.nd: JSON = next((x for x in nd_data if x['Index'] == self.idx), None)
+        self.nm: JSON = next((x for x in nm_data if x['device'] == self.name), None)
+
+        # Filter resolved's DNS data
+        self.dns_addresses: list = None
+        if resolved_data[0]:
+            self.dns_addresses = []
+            for itr in resolved_data[0]:
+                if int(itr[0]) == int(self.idx):
+                    ipfamily = itr[1]
+                    dns = itr[2]
+                    self.dns_addresses.append(socket.inet_ntop(ipfamily, b''.join([v.to_bytes(1, 'big') for v in dns])))
+        self.dns_search: list = None
+        if resolved_data[1]:
+            self.dns_search = []
+            for v in resolved_data[1]:
+                if int(v[0]) == int(self.idx):
+                    self.dns_search.append(str(v[1]))
+
+        # Filter route data
+        _routes: list = []
+        self.routes: list = None
+        if route_data[0]:
+            _routes += route_data[0]
+        if route_data[1]:
+            _routes += route_data[1]
+        if _routes:
+            self.routes = []
+            for obj in _routes:
+                if obj.get('dev') == self.name:
+                    elem = {'to': obj.get('dst')}
+                    val = obj.get('gateway')
+                    if val:
+                        elem['via'] = val
+                    val = obj.get('prefsrc')
+                    if val:
+                        elem['from'] = val
+                    val = obj.get('metric')
+                    if val:
+                        elem['metric'] = val
+                    val = obj.get('type')
+                    if val:
+                        elem['type'] = val
+                    val = obj.get('scope')
+                    if val:
+                        elem['scope'] = val
+                    val = obj.get('protocol')
+                    if val:
+                        elem['protocol'] = val
+                    self.routes.append(elem)
+
+        self.addresses: list = None
+        if 'addr_info' in ip and ip['addr_info']:
```

Comment:
nitpick (no need to change): since python 3.8 this pattern can be simplified a bit with the := operator
```
if addr_info := ip.get('addr_info'):
  for addr in addr_info:
```